### PR TITLE
KWS-3035: Highlighting Comment-Caption input view

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
https://jira.superawesome.tv/browse/KWS-3035

## Description
* Added TaggableTextView
* Added Subclass of the above, TaggableStickerTextView

## Screenshots
simulator screen shot - iphone 7 - 2018-06-06 at 17 36 32
simulator screen shot - iphone 7 - 2018-06-06 at 17 36 57

### CODE REVIEW

File: PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+public protocol StickerButtonDelegate: class {
+    func stickerButtonPressed(withTaggableTextView taggableStickerTextView: TaggableStickerTextView)
+}
+
+public class TaggableStickerTextView: TaggableTextView {
+
+    // MARK: Constants
+    
+    private let stickerButtonSize = CGSize(width: 30.0, height: 30.0)
```
Comment (GuilhermeMota93) [inline PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift]: Maybe magic numbers?

File: PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        super.setUp()
+    }
+    
+    override internal func addSubviews() {
+        super.addSubviews()
+        addSubview(stickerButton)
+    }
+    
+    override internal func addConstriants() {
+        let topPlaceHolderInset: CGFloat = 3.0
```
Comment (GuilhermeMota93) [inline PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift]: Typo in `Constriants`

File: PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        let leadingPlaceHolderInset: CGFloat = 5.0
+        let trailingButtonInset: CGFloat = 4.0
```
Comment (GuilhermeMota93) [inline PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift]: Also, magic numbers?

File: PopJamUI/Classes/GenericViews/TaggableTextView.swift [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        if matchRange.location < selectedRange.location
+            && (matchRange.location + matchRange.length) >= selectedRange.location
+            && matchRange.length >= minimumTagLength {
+            let extractRange = text.index(text.startIndex, offsetBy: matchRange.location)..<text.index(text.startIndex, offsetBy: matchRange.location + matchRange.length)
+            print("searching for tag: \(String(text[extractRange]))")
+            delegate?.requiresAutoComplete(toBeVisible: true, taggableTextView: self)
+            delegate?.requestAutoCompleteSearch(withText: String(text[extractRange]), taggableTextView: self)
+        } else {
+            print("nothing to do")
+            delegate?.requiresAutoComplete(toBeVisible: false, taggableTextView: self)
```
Comment (GuilhermeMota93) [inline PopJamUI/Classes/GenericViews/TaggableTextView.swift]: Do we want to keep the prints? Line 167 & 171

File: PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift (hunk 3 above)
Comment (me2k) [reply PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift]: These are not magic numbers as they are named ;)

File: PopJamUI/Classes/GenericViews/TaggableTextView.swift (hunk 4 above)
Comment (me2k) [reply PopJamUI/Classes/GenericViews/TaggableTextView.swift]: I'll remove those

File: PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift (hunk 2 above)
Comment (me2k) [reply PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift]: fixed :)

File: PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift (hunk 3 above)
Comment (GuilhermeMota93) [reply PopJamUI/Classes/GenericViews/TaggableStickerTextView.swift]: Don't know what I was doing here, apologies 👍

**Pull request merged**